src/shared/istiopolicy/creator.go

```diff
@@ -42,71 +45,110 @@ func NewCreator(client client.Client, recorder *injectablerecorder.InjectableRec
 	}
 }
 
-func (c *Creator) Create(
+func (c *Creator) DeleteAll(
 	ctx context.Context,
-	intents *v1alpha2.ClientIntents,
-	clientIntentsNamespace string,
-	clientServiceAccountName string,
+	clientIntents *v1alpha2.ClientIntents,
 ) error {
-	return c.handleAuthorizationPolicy(ctx, intents, clientIntentsNamespace, clientServiceAccountName)
+	clientFormattedIdentity := v1alpha2.GetFormattedOtterizeIdentity(clientIntents.Spec.Service.Name, clientIntents.Namespace)
+
+	err := c.client.DeleteAllOf(ctx,
+		&v1beta1.AuthorizationPolicy{},
+		client.MatchingLabels{v1alpha2.OtterizeIstioClientAnnotationKey: clientFormattedIdentity})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		c.recorder.RecordWarningEventf(clientIntents, ReasonGettingIstioPolicyFailed, "could not get istio policies: %s", err.Error())
+		return err
+	}
+
+	return nil
 }
 
-func (c *Creator) Delete(
+func (c *Creator) Create(
 	ctx context.Context,
-	intents *v1alpha2.ClientIntents,
+	clientIntents *v1alpha2.ClientIntents,
+	clientNamespace string,
+	clientServiceAccount string,
 ) error {
-	for _, intent := range intents.GetCallsList() {
-		policyName := c.getPolicyName(intents, intent)
-		policy := &v1beta1.AuthorizationPolicy{}
-		err := c.client.Get(ctx, types.NamespacedName{
-			Name:      policyName,
-			Namespace: intent.GetServerNamespace(intents.Namespace)},
-			policy)
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				continue
-			}
+	clientFormattedIdentity := v1alpha2.GetFormattedOtterizeIdentity(clientIntents.Spec.Service.Name, clientNamespace)
+
+	var existingPolicies v1beta1.AuthorizationPolicyList
+	err := c.client.List(ctx,
+		&existingPolicies,
+		client.MatchingLabels{v1alpha2.OtterizeIstioClientAnnotationKey: clientFormattedIdentity})
+	if err != nil {
+		c.recorder.RecordWarningEventf(clientIntents, ReasonGettingIstioPolicyFailed, "could not get istio policies: %s", err.Error())
+		return err
+	}
 
-			logrus.WithError(err).Errorf("could not get istio policy %s", policyName)
-			return err
-		}
+	validPolicies, err := c.handelPolicies(ctx, clientIntents, clientNamespace, clientServiceAccount, existingPolicies)
```

Comment:
handel -> handle

Reply:
Also let's  call `validPolicies` the `newOrUpdatedPolicies` or something that indicates that they're not valid or invalid, but just new or updated.